cmd/soroban-rpc/internal/transactions/transactions_test.go

```diff
@@ -311,6 +348,34 @@ func TestIngestTransactions(t *testing.T) {
 	require.Len(t, store.transactions, 3)
 }
 
+func TestGetTransactionsWithEventData(t *testing.T) {
+	store := NewMemoryStore(interfaces.MakeNoOpDeamon(), "passphrase", 100)
+
+	// Insert ledger 1
+	metaWithEvents := txMetaWithEvents(1, false)
+	require.NoError(t, store.IngestTransactions(metaWithEvents))
+	require.Len(t, store.transactions, 1)
+
+	// check events data
+	tx, ok, _ := store.GetTransaction(txHash(1, false))
+	require.True(t, ok)
+	require.NotNil(t, tx.Events)
+	require.Len(t, tx.Events, 1)
+
+	var event xdr.DiagnosticEvent
+	err := event.UnmarshalBinary(tx.Events[0])
+	require.Nil(t, err)
+	require.Equal(t, xdr.ContractEventTypeContract, event.Event.Type)
+	require.Equal(t, int32(0), event.Event.Body.V)
+
+	require.NotNil(t, event.Event.Body.V0.Topics)
+	require.Equal(t, xdr.ScValTypeScvSymbol, event.Event.Body.V0.Topics[0].Type)
+
+	require.NotNil(t, event.Event.Body.V0.Data)
+	require.Equal(t, xdr.ScValTypeScvSymbol, event.Event.Body.V0.Data.Type)
```

Comment:
I think it would be better to marshal `metaWithEvents.V1.TxProcessing[0].TxApplyProcessing.V3. SorobanMeta.Events` into binary and then compare the resulting binary blob against `tx.Events`. That is going to be a stronger comparison than checking the event types.

Reply:
Okay !! Relying more on `metaWithEvents.V1.TxProcessing[0].TxApplyProcessing.GetDiagnosticEvents()` than `SorobanMeta.Events`